### DISCUSSION TITLE
Issue with FMU

### DISCUSSION THREAD

**Discussion by Amuri44** [OpenModelica/OpenModelica#10631] [Q&A] [answered]
I have a "FMU_Test.mo" and I want to make ".fmu" of it to you use in Matlabl Simulink. 

- First, I tried to generate the FMU by the OMEdit, however, we I import it into Simulink (with FMI Toolbox). I get the error that the DLL could not be loaded.
How to fix that?

- Then, I tried use script as below but I could not see the generated ".fmu" in "C:\Users\am\AppData\Local\Temp\OpenModelica\OMEdit". Where to see it?

```modelica
>> loadModel(Modelica,{"4.0.0"});
>> getErrorString();
>> loadFile("C:/Users/am/OneDrive - dd.com/pathToFile/FMU_Test.mo");
>> convertPackageToLibrary(FMU_Test, Modelica,{"4.0.0"})
{true}
>> buildModelFMU(FMU_Test, version="2.0", fmuType="cs",
              fileNamePrefix="ConstInteger", platforms={"dynamic"}, includeResources=false);
```

**Comment by AnHeuermann** [comment, marked as answer]
`buildModelFMU()` will output the location of the generated FMU. It is shown in the message browser in OMEdit or on the shell where you run your MOS script. If something failed add `getErrorString();` after calling `buildModelFMU()` to see the error messages.
When calling the script it will be generated in your current working directory (the location where you started your shell).

One issue could be, that you have spaces in your path `"C:/Users/am/OneDrive - emtp.com/pathToFile/FMU_Test.mo"`. Linux tools don't always handle spaces very well. Try changing it. If that doesn't help add the full error message to your question.

Another problem is the extra `;` after `buildModelFMU()`. OMShell hides the output of functions when `;` is appended to function calls.

**Reply by Amuri44** [reply]
Thanks for your prompt. I open cmd as below and run OMShell
```modelica
PS C:\Users\OneDrive\Desktop> & "C:\Program Files (x86)\OpenModelica1.22.0-dev-32bit\bin\OMShell.exe"
```

```Modelica
>> loadModel(Modelica,{"4.0.0"});
>> getErrorString();
>> loadFile("C:/Users/am/OneDrive - dd.com/pathToFile/FMU_Test.mo");
"Warning: FMU_Test requested package Modelica of version 3.2.3. Modelica 4.0.0 is used instead which states that it is only compatible with a conversion script. Use convertPackageToLibrary(FMU_Test, Modelica, \"4.0.0\") to run the conversion script or proceed with potential issues as a result.
"
>> convertPackageToLibrary(FMU_Test, Modelica,{"4.0.0"})
{true}
>> buildModelFMU(FMU_Test, version="2.0", fmuType="cs",
              fileNamePrefix="ConstInteger", platforms={"dynamic"}, includeResources=false);
>> getErrorString();
>> 
```
I also tried to put the .mo file in this path to avoid the problems with spaces, but it led to the same issue:

```modelica
>>loadFile("C:/Tes/FMU_Test.mo");
"Warning: FMU_Test requested package Modelica of version 3.2.3. Modelica 4.0.0 is used instead which states that it is only compatible with a conversion script. Use convertPackageToLibrary(FMU_Test, Modelica, \"4.0.0\") to run the conversion script or proceed with potential issues as a result.
```modelica

For reference, please note that the .mod file is located here:
![ee](https://user-images.githubusercontent.com/86628555/236299876-7aa89957-19fb-41eb-a864-ee4ec8d6e78c.png)

**Reply by AnHeuermann** [reply]
Try installing MSL 3.2.3 and skipping the conversion of `FMU_test.mo`  for now.
And I think OMShell hides output when you use `;` after a command. Try removing it.
Otherwise this looks like a bug. Can you open an issue and share your Modelica file to reproduce this?

**Reply by casella** [reply]
> Try installing MSL 3.2.3

I guess you mean "loading". MSL 3.2.3 is installed by default.

**Reply by AnHeuermann** [reply]
Yes, so adding 

```
loadModel(Modelica, {"3.2.3"})
```

to the scripting calls.

**Reply by Amuri44** [reply]
Thanks for your help. Yes, now I can see the generated six files:
.fmu
.txt
.libs
.txt
.makefile
.jbson

However, when I try to load the .fmu into Matlabl Simulink (via  FMU Import in Simulink Extras Library), I got errors:
- Error parsing FMU 'pathtofile/FMUname.fmu'
- Caused by: Error in supplied FMU: Dynamic library file('FMUname.dll) does not exist in FMU 'pathtofile/FMUname.fmu'
Any way to solve it?

**Reply by AnHeuermann** [reply]
> However, when I try to load the .fmu into Matlabl Simulink (via FMU Import in Simulink Extras Library), I got errors:

Ask a new question for this, maybe someone has experience with Matlab FMU import for OpenModelica FMUs.

**Reply by Amuri44** [reply]
Thank you very much for your guide!
